docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+# To plot a 2D basis set, we evaluate the basis on a grid of points over the basis function domain.
+# We use the `evaluate_on_grid` method of neurostatslib.basis for this.
+
+X, Y, Z = prod_basis.evaluate_on_grid(200, 200)
+
+# Setup a 3D plot
+fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+
+# Plot only the support
+Z[Z == 0] = np.nan
+ax.plot_surface(X, Y, Z[50], cmap="viridis", alpha=0.8)
+ax.plot_surface(X, Y, Z[83], cmap="rainbow", alpha=0.8)
+ax.plot_surface(X, Y, Z[125], cmap="inferno", alpha=0.8)
+plt.title(f"Product basis with {eval_basis.shape[0]} elements")
+plt.show()
+
+# %%
+# Define and Evaluate a 2D Additive Basis
+# ---------------------------------------
+# Just like in the multiplicative case, we can create an additive basis function as the sum of two 1D basis functions.
+# We approximate the response function for multi-dimensional stimuli as the sum of two 1D basis functions.
+# For the 2D case, we can approximate our response function as follows:
+# $$
+# f(x,y) \\approx \alpha_i \cdot a_i(x) + \beta_j \cdot b_j(y)
+# $$
+# Here, we simply add two basis objects, 'a_basis' and 'b_basis', together to get the additive basis.
+
+
+additive_basis = a_basis + b_basis
+
+# Evaluate the basis functions for the given trajectory.
+eval_basis = additive_basis.evaluate(x_coord, y_coord)
+
+print(f"Sum of two 1D splines with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- a_basis had {a_basis._n_basis_funcs} elements\n\t- b_basis had {b_basis._n_basis_funcs} elements.")
+
+# %%
+# Plot the Additive Basis Elements
+# --------------------------
+# To plot a 2D additive basis set, we need to evaluate the basis on a grid of points over the basis function domain.
+# This can be done using the `neurostatslib.basis.evaluate_on_grid` method.
+
+X, Y, Z = additive_basis.evaluate_on_grid(200, 200)
+
+fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+
+# Only plot the support
+Z[Z == 0] = np.nan
+ax.plot_surface(X, Y, Z[5], cmap="viridis", alpha=0.8)
+ax.plot_surface(X, Y, Z[25], cmap="inferno", alpha=0.8)
+plt.title(f"Additive basis with {eval_basis.shape[0]} elements")
+plt.show()
+
+# %%
+# Combining Basis Types
+# ---------------------
+# Basis objects of different types can be combined through multiplication or addition.
+# This feature is particularly beneficial when one of the axes represents a periodic variable
+# such as an angle. A practical example would be characterizing responses to position
+# in a linear maze and the LFP phase angle.
+
+c_basis = nsl.basis.RaisedCosineBasisLinear(n_basis_funcs=8)
+
+# Multiply or add basis of different types
+prod_basis = b_basis * c_basis
+add_basis = b_basis + c_basis
+
+print(f"Product of two 1D splines of different type with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- b_basis is of type {b_basis.__class__}\n\t- c_basis is of type {c_basis.__class__}")
+
```

Comment:
It is not clear to me how the code here relates to the text. Is this an angular basis? I think what you're showing is that the two combined bases don't need to be the same subclass, but that's not clear. I think that can just be said (doesn't need its own block and example code).

I think the way to do that is with [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types)! The `info` type probably. That's supported by [mkdocs gallery](https://smarie.github.io/mkdocs-gallery/generated/gallery/plot_0_sin/)